### PR TITLE
HTTP proxy support for Rancher Continuous Delivery (fleet)

### LINKED ISSUE #30042: Add HTTP proxy support for Continuous Delivery (Fleet)
**What kind of request is this (question/bug/enhancement/feature request):**
`enhancement`

**Steps to reproduce (least amount of steps as possible):**

Prerequisite: Rancher is deployed in on-prem environment behind the HTTP proxy, with Continuous Delivery feature (Fleet) enabled.
In Cluster Manager / Continuous Delivery / Git Repos, register new Git Repo pointing to external git repository, e.g.:

- Name: `podinfo`
- Repository URL: `https://github.com/stefanprodan/podinfo.git`
- Paths: `/charts/podinfo`
- Deploy To: pick target cluster

**Result:**

It is expected that application from registered "podinfo" GitRepo is deployed to desired cluster.
However, application is not deployed, "podinfo" GitRepo object is stuck in "Git Updating" state, and its `status:` property contains the following message:
>  `Repo [https://github.com/stefanprodan/podinfo.git] is not accessible:
    >  Get "https://api.github.com/repos/stefanprodan/podinfo/commits/master": context
      deadline exceeded (Client.Timeout exceeded while awaiting headers)`

**Other details that may be helpful:**

Following issues have been identified and addressed after investigation:

- [gitjob controller](https://github.com/rancher/gitjob), used by Fleet, doesn't support HTTP proxy settings, therefore it can't clone external git repositories  when deployed behind the proxy. This issue is addressed by https://github.com/rancher/gitjob/pull/20
- Fleet helm chart doesn't have support for HTTP proxy settings. Addressed by https://github.com/rancher/fleet/pull/143
- Although Rancher supports HTTP proxy settings, it doesn't pass them to Fleet deployment bundled with Rancher. Addressed by https://github.com/rancher/rancher/pull/29936 

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): `v2.5.2`
- Installation option (single install/HA): HA

### PR DESCRIPTION
**Pull request by somaritane**
This PR, together with https://github.com/rancher/fleet/pull/143 and https://github.com/rancher/gitjob/pull/20, helps to propagate Rancher HTTP proxy settings to fleet system chart installation, which enables fleet to work with external git repositories in environments behind corporate proxy.
Fixes #30042

### CODE REVIEW

**Comment by StrongMonkey** [comment]
@somaritane Can you create a issue in rancher to track this? Thanks a lot for the pull request!

**Comment by somaritane** [comment]
> @somaritane Can you create an issue in rancher to track this? Thanks a lot for the pull request!

@StrongMonkey, sure thing, I've created https://github.com/rancher/rancher/issues/30042

File: pkg/controllers/dashboard/fleetcharts/controller.go [hunk 1]
```diff
@@ -61,10 +62,25 @@ func (h *handler) onSetting(key string, setting *v3.Setting) (*v3.Setting, error
...
+		"apiServerCA":  settings.CACerts.Get(),
+		"global":       systemGlobalRegistry,
+	}
+
+	gitjobChartValues := map[string]interface{}{}
+
+	if envVal, ok := os.LookupEnv("HTTP_PROXY"); ok {
+		gitjobChartValues["proxy"] = envVal
+	}
+	if envVal, ok := os.LookupEnv("NO_PROXY"); ok {
```
Comment (StrongMonkey) [inline pkg/controllers/dashboard/fleetcharts/controller.go:77]: Did you miss `HTTPS_PROXY`?

Comment (StrongMonkey) [reply pkg/controllers/dashboard/fleetcharts/controller.go:77]: Nvm, I think they have the same setting.

**Review by StrongMonkey — APPROVED** [review]

File: pkg/controllers/dashboard/fleetcharts/controller.go (hunk 1 above)
Comment (somaritane) [reply pkg/controllers/dashboard/fleetcharts/controller.go:77]: Good point, technically they can be different.

Comment (StrongMonkey) [reply pkg/controllers/dashboard/fleetcharts/controller.go:80]: @somaritane https://github.com/rancher/rancher/blob/362fb20d9e8ef0e2c779e1ee84300e8fda9eb316/chart/templates/deployment.yaml#L91 they will always have the same setting due to the configuration in our charts. Can you revert the new added changes? It should be fine

**Comment by somaritane** [comment]
@StrongMonkey, can you please help me with drone pipeline restart, looks like some integration tests are intermittently failing for one of `default-linux-amd64` targets, and I don't have permissions there.

**Review by StrongMonkey — APPROVED** [review]

File: pkg/controllers/dashboard/fleetcharts/controller.go [hunk 2]
```diff
@@ -61,10 +62,25 @@ func (h *handler) onSetting(key string, setting *v3.Setting) (*v3.Setting, error
...
-			"global":       systemGlobalRegistry,
-		})
+
+	fleetChartValues := map[string]interface{}{
+		"apiServerURL": settings.ServerURL.Get(),
+		"apiServerCA":  settings.CACerts.Get(),
+		"global":       systemGlobalRegistry,
+	}
+
+	gitjobChartValues := map[string]interface{}{}
```
Comment (dramich) [inline pkg/controllers/dashboard/fleetcharts/controller.go:72]:
```suggestion
	gitjobChartValues :=  make(map[string]interface{})
```
Use `make` when creating a new empty map

**Comment by StrongMonkey** [comment]
@somaritane Format error

**Comment by somaritane** [comment]
@StrongMonkey thanks, amended.

**Review by dramich — APPROVED** [review]

**Comment by somaritane** [comment]
@dramich , @StrongMonkey thank you for help, pleasure to collaborate with you!

**Pull request merged**